File: frontend/src/currency.ts
export type Currency = 'USD' | 'INR' | 'EUR' | 'JPY' | 'GBP' | 'CNY'

export const RATES: Record<Currency, number> = {
  USD: 1,
  INR: 84,
  EUR: 0.92,
  JPY: 150,
  GBP: 0.78,
  CNY: 7.2,
}

export const SYMBOLS: Record<Currency, string> = {
  USD: '$',
  INR: '₹',
  EUR: '€',
  JPY: '¥',
  GBP: '£',
  CNY: '¥',
}

export const CURRENCIES: { code: Currency; label: string }[] = [
  { code: 'USD', label: 'USD ($)' },
  { code: 'INR', label: 'INR (₹)' },
  { code: 'EUR', label: 'EUR (€)' },
  { code: 'JPY', label: 'JPY (¥)' },
  { code: 'GBP', label: 'GBP (£)' },
  { code: 'CNY', label: 'CNY (¥)' },
]

export function toBase(amount: number, currency: Currency): number {
  return amount / RATES[currency];
}

export function formatCurrency(amount: number | undefined, currency: Currency): string {
  if (amount === undefined) return "-";
  return `${SYMBOLS[currency]}${(amount * RATES[currency]).toFixed(2)}`;
}

File: mobile/lib/utils/currency.dart
enum AppCurrency {
  usd('USD', '\$', 1.0, 'USD (\$)'),
  inr('INR', '₹', 84.0, 'INR (₹)'),
  cny('CNY', '¥', 7.2, 'CNY (¥)');

  final String code;
  final String symbol;
  final double rate;
  final String label;

  const AppCurrency(this.code, this.symbol, this.rate, this.label);

  double toBase(double amount) => amount / rate;

  double fromBase(double amount) => amount * rate;

  String format(double? amount) {
    if (amount == null) return '-';
    final converted = amount * rate;
    return '$symbol${converted.toStringAsFixed(2)}';
  }
}
